model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/DesignElementHandle.java

```diff
@@ -1054,10 +1051,9 @@ public PropertyHandle getPropertyHandle(String propName) {
 			PropertyHandle propHandle = propHandles.get(propName);
 			if (propHandle != null) {
 				return propHandle;
-			} else {
-				cachePropertyHandles();
-				return propHandles.get(propName);
 			}
+			cachePropertyHandles();
+			return propHandles.get(propName);
```

Comment:
There the `propHandle` variable already has already gotten the value.

Reply:
I solved only the warning to make small changes not to reenigeer the variables.

Reply:
Yes, this was already poor.  When touching it, might as well make it less poor.  But that's your choice.

Reply:
I confirm your thinking.

But the "cachePropertyHandles" can change the propHandles and therefore the get-Method will be use again instead of the "propHandle". So for me it's ok, what would the improvement from your side?

Reply:
So it might be a different object than what was in the map before?  In that case it’s surprising but fine.